Character controller

ElvenNurse opened the pull request.

ElvenNurse committed:
Added character, episode, location models
Added character, episode, location repositories
Added character, episode, location services
Added HttpClient
Added cron update controller
Added character controller

ElvenNurse committed:
Fixed imports

Tarabarnik commented on src/main/java/com/example/rickmortiapi/controller/CharacterController.java:
```diff
@@ -0,0 +1,34 @@
+@RestController
+@RequestMapping("/character")
+public class CharacterController {
+    private UpdateDatabaseController updateDatabaseController;
```
не думаю, що варто викликати контролер в кортролері. можливо, варто зробити UpdateDatabaseController сервісом?

ElvenNurse replied:
It was there for testing, forgot to remove))

Tarabarnik commented on src/main/java/com/example/rickmortiapi/service/EpisodeServiceImpl.java:
```diff
@@ -0,0 +1,74 @@
+        }
+    }
+
+    private void mapAndSaveEpisodes(List<ApiEpisodeDto> episodeDtos) {
```
And означає, що метод порушує SRP

Tarabarnik commented on src/main/java/com/example/rickmortiapi/service/LocationServiceImpl.java:
```diff
@@ -0,0 +1,74 @@
+        }
+    }
+
+    private void mapAndSaveLocations(List<ApiLocationDto> locationDtos) {
```
same

Tarabarnik commented on src/main/java/com/example/rickmortiapi/util/HttpClient.java:
```diff
@@ -0,0 +1,31 @@
+        return null;
+    }
+
+}
```
додай пустий рядок

ElvenNurse committed:
Removed unused field in CharacterController
Slightly reworked updateRepository methods in Character/Episode/Location services

ElvenNurse committed:
Added Swagger support
Added index.html page
Added added Maven Checkstyle plugin
Small fixes
